back/scripts/enrichment/marches_enricher.py

```diff
@@ -20,5 +20,77 @@ def get_input_paths(cls, main_config: dict) -> typing.List[Path]:
     @classmethod
     def _clean_and_enrich(cls, inputs: typing.List[pl.DataFrame]) -> pl.DataFrame:
         marches, *_ = inputs
-        # Data analysts, please add your code here!
+        # TODO casting temporaire de titulaire_id en String, il faudra modifier le fichier source
+        marches = marches.with_columns(
+            pl.col("titulaire_id").cast(pl.String).alias("titulaire_id")
+        )
+
+        marches = cls.type_identifiant_titulaire_enrich(marches)
+        return marches
+
+    @staticmethod
+    def type_identifiant_titulaire_enrich(marches: pl.DataFrame) -> pl.DataFrame:
+        """
+        Normalize titulaire_typeIdentifiant column from titulaire_typeIdentifiant and titulaire_id.
+        """
+
+        # TODO : Il y a encore environ 600 titulaire_typeIdentifiant avec des titulaire_id non null qui sont null
+        SIRET_REGEX = r"^\d{14}$"  # 14 chiffres uniquement
+        SIREN_REGEX = r"^\d{9}$"  # 9 chiffres uniquement
+        TVA_REGEX = r"^[A-Z]{2}\d{9,12}$"  # Ex: GB123456789 ou FR12345678912
+
+        marches = (
+            marches.with_columns(
+                pl.when(pl.col("titulaire_typeIdentifiant") == "HORS_UE")
+                .then(pl.lit("HORS-UE"))
+                .otherwise(pl.col("titulaire_typeIdentifiant"))
+                .alias("titulaire_typeIdentifiant")
+            )
+            .with_columns(
+                pl.when(pl.col("titulaire_typeIdentifiant") == "TVA_INTRACOMMUNAUTAIRE")
+                .then(pl.lit("TVA"))
+                .otherwise(pl.col("titulaire_typeIdentifiant"))
+                .alias("titulaire_typeIdentifiant")
+            )
+            .with_columns(
+                pl.when(pl.col("titulaire_typeIdentifiant") == "FRW")
+                .then(pl.lit("FRWF"))
+                .otherwise(pl.col("titulaire_typeIdentifiant"))
+                .alias("titulaire_typeIdentifiant")
+            )
+            .with_columns(
+                pl.when(pl.col("titulaire_typeIdentifiant") == "UE")
+                .then(pl.lit("TVA"))
+                .otherwise(pl.col("titulaire_typeIdentifiant"))
+                .alias("titulaire_typeIdentifiant")
+            )
```

Comment:
on peut rendre l'ensemble un peu plus court avec `map_dict` :) ! 
```suggestion
    mapping = {
        "HORS_UE": "HORS-UE",
        "TVA_INTRACOMMUNAUTAIRE": "TVA",
        "FRW": "FRWF",
        "UE": "TVA",
    }

    marches = marches.with_columns(
        pl.col("titulaire_typeIdentifiant")
        .map_dict(mapping, default=pl.col("titulaire_typeIdentifiant"))
    )
```

Reply:
C'est bon